### PR TITLE
Add dockerhub authentication for kubelet (v8)

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/13981

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: pkg/template/worker_template.go [hunk 1]
```diff
@@ -365,6 +365,12 @@ storage:
       contents:
         source: "data:text/plain;charset=utf-8;base64,{{  index .Files "conf/docker-daemon.json" }}"
 
+    - path: /root/.docker/config.json
+      filesystem: root
+      mode: 0644
+      contents:
+        source: "data:text/plain;charset=utf-8;base64,{{  index .Files "conf/kubelet-docker-config.json" }}"
+
```
Comment (calvix) [inline pkg/template/worker_template.go:373]: probabbly stupid question, isnt docker running as docker user? so why it would take creds from root?

Comment (wtty-fool) [reply pkg/template/worker_template.go:373]: kubelet is using `/root/.docker/config.json`

Comment (corest) [reply pkg/template/worker_template.go:373]: There is no even sunch user as docker

Comment (calvix) [reply pkg/template/worker_template.go:373]: ah right, it's kubelet pulling the images, not docker, sorry ;)

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,10 @@ and this project's packages adheres to [Semantic Versioning](http://semver.org/s
```
Comment (kopiczko) [inline CHANGELOG.md:7]: It isn't quite on par with https://github.com/giantswarm/k8scloudconfig/pull/818/files

E.g. `DockerhubToken` validation is missing. Isn't this cherry-pick'able?

**Review by kopiczko — APPROVED** [review]

**Pull request merged**